sys/net/gnrc/routing/ipv6_auto_subnets/gnrc_ipv6_auto_subnets.c

```diff
@@ -0,0 +1,221 @@
+                             const ipv6_addr_t *prefix, uint8_t bits,
+                             uint8_t idx, uint8_t idx_bits)
+{
+    uint8_t bytes = bits / 8;
+    uint8_t rem   = bits % 8;
+    int8_t shift  = 8 - rem - idx_bits;
+
+    /* first copy old prefix */
+    memcpy(out, prefix, bytes);
+
+    /* make sure unused bits are cleared */
+    if (rem) {
+        uint8_t mask = 0xff << (8 - rem);
+        out->u8[bytes] = prefix->u8[bytes] & mask;
+    } else {
+        out->u8[bytes] = 0;
+    }
+
+    /* if new bits are between bytes, first copy over the most significant bits */
+    if (shift < 0) {
+        out->u8[bytes] |= idx >> -shift;
+        out->u8[++bytes] = 0;
+        shift += 8;
+    }
+
+    /* shift remaining bits at the end of the prefix */
+    out->u8[bytes] |= idx << shift;
+}
+
+static bool _remove_old_prefix(gnrc_netif_t *netif,
+                               const ipv6_addr_t *pfx, uint8_t pfx_len,
+                               gnrc_pktsnip_t **ext_opts)
+{
+    gnrc_ipv6_nib_pl_t entry;
+    gnrc_pktsnip_t *tmp;
+    void *state = NULL;
+    ipv6_addr_t old_pfx;
+    uint8_t old_pfx_len = 0;
+
+    /* iterate prefix list to see if the prefix already exists */
+    while (gnrc_ipv6_nib_pl_iter(netif->pid, &state, &entry)) {
+        uint8_t match_len = ipv6_addr_match_prefix(&entry.pfx, pfx);
+
+        /* The prefix did not change - nothing to do here */
+        if (match_len >= pfx_len && pfx_len == entry.pfx_len) {
+            return true;
+        }
+
+        /* find prefix that is closest to the new prefix */
+        if (match_len > old_pfx_len) {
+            old_pfx_len = entry.pfx_len;
+            old_pfx = entry.pfx;
+        }
+    }
+
+    /* no prefix found */
+    if (old_pfx_len == 0) {
+        return false;
+    }
+
+    DEBUG("auto_subnets: remove old prefix %s/%u\n",
+          ipv6_addr_to_str(addr_str, &old_pfx, sizeof(addr_str)), old_pfx_len);
+
+    /* invalidate old prefix in RIO */
+    tmp = gnrc_ndp_opt_ri_build(&old_pfx, old_pfx_len, 0,
+                                NDP_OPT_RI_FLAGS_PRF_NONE, *ext_opts);
+    if (tmp) {
+        *ext_opts = tmp;
+    }
+
+    /* remove the prefix */
+    gnrc_ipv6_nib_pl_del(netif->pid, &old_pfx, old_pfx_len);
+
+    return false;
+}
+
+static void _configure_subnets(uint8_t subnets, gnrc_netif_t *upstream,
+                               const ndp_opt_pi_t *pio)
+{
+    gnrc_netif_t *downstream = NULL;
+    gnrc_pktsnip_t *ext_opts = NULL;
+    const ipv6_addr_t *prefix = &pio->prefix;
+    uint32_t valid_ltime = byteorder_ntohl(pio->valid_ltime);
+    uint32_t pref_ltime = byteorder_ntohl(pio->pref_ltime);
+    const uint8_t prefix_len = pio->prefix_len;
+    uint8_t new_prefix_len, subnet_len;
+
+    DEBUG("auto_subnets: create %u subnets\n", subnets);
+
+    /* Calculate remaining prefix length. For n subnets we consume ⌊log₂ n⌋ + 1 bits.
+     * To calculate ⌊log₂ n⌋ quickly, find the position of the most significant set bit
```

Comment:
Not sure if it is wise to use unicode here (especially since it can easily expressed without:

```suggestion
    /* Calculate remaining prefix length. For n subnets we consume floor(log_2 n) + 1 bits.
     * To calculate floor(log_2 n) quickly, find the position of the most significant set bit
```

Reply:
Besides having Unicode here in the first place: At first sight I also confused the flooring-brackets `⌊⌋` with normal brackets `[]` because I have a very small font-size config. Then I thought something is wrong with the display and only then I realized you where using Unicode here :stuck_out_tongue_winking_eye:. I think among programers `floor()` in general is also better understood than `⌊⌋`